misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java

```diff
@@ -139,22 +140,22 @@ public static JsonToRecordResponse convert(String jsonString, String recordName,
      * @param openApi OpenAPI model
      * @param isRecordTypeDescriptor To denote the record, a type descriptor
      * @param isClosedRecord Denotes the record, a closed record
-     * @return {@link ArrayList}  List of Record Nodes
+     * @return {@link Map}  Map of Record Nodes
      * @throws JsonToRecordConverterException In case of bad record fields
      */
-    private static ArrayList<NonTerminalNode> generateRecords(OpenAPI openApi, boolean isRecordTypeDescriptor,
+    private static Map<String, NonTerminalNode> generateRecords(OpenAPI openApi, boolean isRecordTypeDescriptor,
                                                               boolean isClosedRecord)
             throws JsonToRecordConverterException {
-        List<NonTerminalNode> typeDefinitionNodeList = new LinkedList<>();
+        Map<String, NonTerminalNode> typeDefinitionNodes = new LinkedHashMap<>();
 
         if (openApi.getComponents() == null) {
-            return new ArrayList<>(typeDefinitionNodeList);
+            return new LinkedHashMap<>(typeDefinitionNodes);
         }
 
         Components components = openApi.getComponents();
 
         if (components.getSchemas() == null) {
-            return new ArrayList<>(typeDefinitionNodeList);
+            return new LinkedHashMap<>(typeDefinitionNodes);
         }
```

Comment:
You can merge these two if clauses together

Reply:
Addressed in https://github.com/ballerina-platform/ballerina-lang/pull/34525/commits/9129a28d18a2ad3bae9e19515f274b8ae939c033